File: 프로그래머스/0/120815. 피자 나눠 먹기 （2）/피자 나눠 먹기 （2）.py
def solution(n):
    def gcd(p, q):
        # 1 <= n <= 6
        if p <= 6:
            q, p = p, q
            if p == 0:
                return q
            return gcd(p, q % p)
        else:
            # base condition
            if q == 0:
                return p
            return gcd(q, p % q)
    return n // gcd(n, 6)

File: 프로그래머스/0/120815. 피자 나눠 먹기 （2）/피자 나눠 먹기 （2）.cpp
#include <string>
#include <vector>

using namespace std;

int gcd(int p, int q)
{
    if (q == 0)
        return p;
    return gcd(q, p % q);
}

int solution(int n)
{
    return n / int (gcd(n, 6));
}